src/SharedInfrastructure/ThrowHelper.cs

```diff
@@ -0,0 +1,134 @@
+// Copyright (c) Six Labors and contributors.
+// Licensed under the Apache License, Version 2.0.
+
+using System;
+using System.Runtime.CompilerServices;
+
+namespace SixLabors
+{
+    /// <summary>
+    /// Helper methods to throw exceptions
+    /// </summary>
+    internal static class ThrowHelper
+    {
+        /// <summary>
+        /// Throws an <see cref="ArgumentNullException"/> when <see cref="Guard.NotNull{TValue}"/> fails.
+        /// </summary>
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        public static void ThrowArgumentNullExceptionForNotNull(string name)
+        {
+            ThrowArgumentNullException(name, $"Parameter \"{name}\" must be not null.");
+        }
+
+        /// <summary>
+        /// Throws an <see cref="ArgumentException"/> when <see cref="Guard.NotNullOrWhiteSpace"/> fails.
+        /// </summary>
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        public static void ThrowArgumentExceptionForNotNullOrWhitespace(string value, string name)
+        {
+            if (value is null)
+            {
+                ThrowArgumentNullException(name, $"Parameter \"{name}\" must be not null.");
+            }
+            else
+            {
+                ThrowArgumentException(name, $"Parameter \"{name}\" must not be empty or whitespace.");
+            }
+        }
+
+        /// <summary>
+        /// Throws an <see cref="ArgumentOutOfRangeException"/> when <see cref="Guard.MustBeLessThan{TValue}"/> fails.
+        /// </summary>
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        public static void ThrowArgumentOutOfRangeExceptionForMustBeLessThan<T>(T value, T max, string name)
+            where T : IComparable<T>
```

Comment:
We don't need the `where T : IComparable<T>` constraint here. I'm afraid it will have a negative effect on code generation.

Reply:
Fixed in https://github.com/SixLabors/SharedInfrastructure/pull/7/commits/e61bb6f47f7fb55d88579f9df15f837244d04260.